lib/Controller/ObjectsController.php

```diff
@@ -280,11 +338,20 @@ public function destroy(int $id): JSONResponse
     {
         // Create a log entry
         $oldObject = $this->objectEntityMapper->find($id);
-        $this->auditTrailMapper->createAuditTrail(old: $oldObject);
-
-        $this->objectEntityMapper->delete($this->objectEntityMapper->find($id));
+        
+        // Clone the object to pass as the new state
+        $newObject = clone $oldObject;
+        $newObject->delete();
+        
+        // Update the object in the mapper instead of deleting
+        $this->objectEntityMapper->update($newObject);
 
-        return new JSONResponse([]);
+        // Create an audit trail with both old and new states
+        $this->auditTrailMapper->createAuditTrail(old: $oldObject, new: $newObject);
```

Comment:
Are the old and new objects not the same in this scenario?

Reply:
No, the newObject is now tagged as deleted.